src/utilities.jl

```diff
@@ -203,14 +203,16 @@ function weighted_color_mean(w1::Real, c1::C, c2::C) where {Cb <: Union{HSV, HSL
                                                             C <: Union{Cb, AlphaColor{Cb}, ColorAlpha{Cb}}}
     normalize_hue(_weighted_color_mean(w1, c1, c2))
 end
-function weighted_color_mean(w1::Real, c1::Gray{Bool}, c2::Gray{Bool})
-    # weighting of two Gray{Bool} would return different color type and therefore omitted
-    throw(DomainError())
-end
 function _weighted_color_mean(w1::Real, c1::Colorant, c2::Colorant)
-    weight1 = convert(promote_type(eltype(c1), eltype(c2)),w1)
-    weight2 = weight1 >= 0 && weight1 <= 1 ? oftype(weight1,1-weight1) : throw(DomainError())
-    mapc((x,y)->weight1*x+weight2*y, c1, c2)
+    @fastmath min(w1, oneunit(w1) - w1) >= zero(w1) || throw(DomainError(w1, "`w1` must be in [0, 1]"))
+    T = promote_type(eltype(c1), eltype(c2))
+    weight1 = convert(T, w1) # TODO: Consider the need for this
+    weight2 = oneunit(weight1) - weight1
+    mapc((x, y)->convert(T, muladd(weight1, x, weight2 * y)), c1, c2)
+end
+function _weighted_color_mean(w1::Integer, c1::C, c2::C) where C <: Colorant
```

Comment:
Any reason not allowing different types of `c1` and `c2`, since it only requires `w1` to be `0` or `1`.

```suggestion
function _weighted_color_mean(w1::Integer, c1::Colorant, c2::Colorant)
```

Reply:
Any reason allowing different types of `c1` and `c2`? They cannot be optimized, i.e. can throw errors.

Reply:
I don't get it, why would it throw errors? Is that because `c1` and `c2` are not promoted to a common type?

Reply:
> Is that because c1 and c2 are not promoted to a common type?

Yes. Of course, no error should be thrown for colors in the same color space, but this optimization is not available.

```julia
julia> weighted_color_mean(1, Gray{Float32}(0), Gray{Float64}(1))
Gray{Float64}(0.0)
```